### ISSUE TITLE
BUG: Backup modules created with --module-only

### ISSUE THREAD

**Issue by shahzebsiddiqui** [easybuilders/easybuild-framework#2441]
@damianam @boegel 

Looks like whenever I want to run --module-only I get a backup module created that is part of ``module av``

The backup should be hidden but I assume you had to specify ``--backup-modules`` in order to do this. Right now it is doing by default.

I am running Lmod 6.5.1 on Redhat 7.3

```
[siddis14@amrndhl1157 config]$ eb M4-1.4.17.eb --module-only --rebuild
== temporary log file in case of crash /hpc/grid/hpcws/hpcswbuild/easybuild/IvyBridge/redhat/7.3/tmp/eb-4tZ2m3/easybuild-2j68H7.log
== processing EasyBuild easyconfig /lustre/workspace/home/hpcswbuild/siddis14/easybuild-easyconfigs/easybuild/easyconfigs/m/M4/M4-1.4.17.eb
== building and installing M4/1.4.17...
== fetching files [skipped]
== creating build dir, resetting environment...
== backup of existing module file stored at /nfs/grid/software/easybuild/IvyBridge/redhat/7.3/modules/all/M4/1.4.17.lua.bak_20180314160540
== unpacking [skipped]
== patching [skipped]
== preparing...
== configuring [skipped]
== building [skipped]
== testing [skipped]
== installing [skipped]
== taking care of extensions [skipped]
== postprocessing [skipped]
== sanity checking [skipped]
== cleaning up [skipped]
== creating module...
== comparing module file with backup /nfs/grid/software/easybuild/IvyBridge/redhat/7.3/modules/all/M4/1.4.17.lua.bak_20180314160540; no differences found
== permissions [skipped]
== packaging [skipped]
== COMPLETED: Installation ended successfully
== Results of the build can be found in the log file(s) /nfs/grid/software/easybuild/IvyBridge/redhat/7.3/software/M4/1.4.17/easybuild/easybuild-M4-1.4.17-20180314.160541.log
== Build succeeded for 1 out of 1
== Temporary log file(s) /hpc/grid/hpcws/hpcswbuild/easybuild/IvyBridge/redhat/7.3/tmp/eb-4tZ2m3/easybuild-2j68H7.log* have been removed.
[siddis14@amrndhl1157 config]$ ml av M4/1.4.17
------------------------------------------------------------------- /nfs/grid/software/easybuild/IvyBridge/redhat/7.3/modules/all --------------------------------------------------------------------
   M4/1.4.17.bak_20180314160540    M4/1.4.17
```

**Comment by boegel** [comment]
@damianam Seems like we should always make the backup modules hidden by using a leading dot (`.`) in the filename?

@rtmclay Is it possible that Lmod 7 only considers files ending in `.lua` as module files, while Lmod 6 was picking up anything that has `.lua` somewhere in the filename?

See above: the (backup) module file `M4/1.4.17.lua.bak_20180314160540` is being shown by Lmod 6.5.1 as `M4/1.4.17.bak_20180314160540` in the output of `module avail`

**Comment by rtmclay** [comment]
It is true that Lmod 7 only considers files that end in *.lua as Lua modulefiles.  Lmod 6.5.1 considers files that have .lua anywhere as a lua modulefile.

**Comment by boegel** [comment]
@damianam Are you up for looking into a fix for this?

**Comment by boegel** [comment]
Fix implemented in #2463, let's follow up there...
